src/main/java/org/alliancegenome/curation_api/model/document/builders/HTPDatasetDocumentBuilder.java

```diff
@@ -0,0 +1,142 @@
+package org.alliancegenome.curation_api.model.document.builders;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.alliancegenome.curation_api.model.document.es.HTPDatasetSearchResultDocument;
+import org.alliancegenome.curation_api.model.entities.HTPExpressionDatasetAnnotation;
+import org.alliancegenome.curation_api.model.entities.HTPExpressionDatasetSampleAnnotation;
+
+import com.okta.commons.lang.Collections;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class HTPDatasetDocumentBuilder {
+
+	public static HTPDatasetSearchResultDocument buildSearchResultDocument(
+			HTPExpressionDatasetAnnotation htpDatasetAnnotation) {
+		HTPDatasetSearchResultDocument doc = new HTPDatasetSearchResultDocument();
+
+		List<HTPExpressionDatasetSampleAnnotation> sampleAnnots = htpDatasetAnnotation.getHtpExpressionDataset()
+				.getHtpExpressionDatasetSampleAnnotation();
+
+		if (htpDatasetAnnotation.getDataProvider() != null) {
+			String dataProvider = htpDatasetAnnotation.getDataProvider().getAbbreviation();
+			doc.setDataProvider(dataProvider);
+		}
+
+		if (htpDatasetAnnotation.getName() != null) {
+			doc.setName(htpDatasetAnnotation.getName());
+			doc.setNameKey(htpDatasetAnnotation.getName());
```

Comment:
Not sure on these... but you might want to check the docs in the public elastic search to make sure the "names" are the same?

Reply:
Yeah, I thought it was weird too, but all of the examples I've looked at appear to be the same.